### ISSUE TITLE
Getting extruded building object

### ISSUE THREAD

**Issue by gizm0bill** [heremaps/harp.gl#1286]
Hi, is there is any possibility to get the actual extruded feature building as an object reference right now?
Something the same as https://developer.here.com/documentation/android-premium/dev_guide/topics/map-extruded-buildings.html 
I want to get the building faces and add some textures on them.

**Comment by zbigg** [comment]
Hi @gizm0bill, thanks for interest in our project.

TLDR, some parts of what you seek is possible but far from easy.

Problem 1.

Due to performance reasons, by default we don't keep any information about identity of map features, so what we see on map is just 3d geometry divided into *tile*s with no knowledge which polygon belongs to which "building" in your case.

To retrieve this information, you must at least tell `OmvDataSource` to `gatherFeatureAttributes` by setting it to `true` in constructor.
Note - this greatly increases memory usage.

See object picking examples: https://www.harp.gl/docs/master/examples/#object-picking.html

Problem 2

Due to already mentioned tiling, one "real" building may end up being split to several tiles and thus rendered by several distinct `THREE.Objects`.
So, we need to correlate "features" from many tiles and this is only available in some data providers.
Not all datasources provide `feature id` for all features.

Namely
 - object picking example using `https://xyz.api.here.com/tiles/herebase.02` shows that you don't have disctinct `$id` for each feature in this case
 - XYZ OSM datasource `https://xyz.api.here.com/tiles/osmbase/256/all`- used in [other example](https://www.harp.gl/docs/master/examples/codebrowser.html?src=src%2Fdatasource_xyzmvt.ts) provides proper feature `$id`s AFAIR.

Problem 3

This is going hardcore as most of API is not documented as good as we would like to

Having `gatherFeatureAttributes` set, you can traverse scene and find all objects with `userData.feature` containing `TileFeatureData` objects to search for buffer geometry segments that are responsible for rendering feature by inspecting `TileFeatureData.objInfos` and correlating it with `TileFeatureData.starts` which tell where in particular `BufferGeometry` given feature starts.

Knowing ranges, you can use [multi material/groups feature](https://threejs.org/docs/#api/en/core/BufferGeometry.groups) of `THREE.Mesh` to render selected ranges with other material - i would start with different color.
 
Problem 4

I'm not certain if we generate usable texture coordinates buffer.

Summarizing this part of this can be done, but require very intimate knowledge of engine.
Unfortunately we don't have ready any ready example to be shown at hand.

Hope that it helps, we're open to questions.

**Comment by gizm0bill** [comment]
Much appreciated!
